cmd/collectors/rest/plugins/volumeanalytics/volumeanalytics.go

```diff
@@ -0,0 +1,289 @@
+package volumeanalytics
+
+import (
+	"fmt"
+	goversion "github.com/hashicorp/go-version"
+	"github.com/netapp/harvest/v2/cmd/poller/plugin"
+	"github.com/netapp/harvest/v2/cmd/tools/rest"
+	"github.com/netapp/harvest/v2/pkg/conf"
+	"github.com/netapp/harvest/v2/pkg/errs"
+	"github.com/netapp/harvest/v2/pkg/matrix"
+	"github.com/netapp/harvest/v2/pkg/util"
+	"github.com/tidwall/gjson"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const explorer = "volume_analytics"
+
+var MaxDirCollectCount = "100"
+
+type VolumeAnalytics struct {
+	*plugin.AbstractPlugin
+	currentVal int
+	client     *rest.Client
+	data       map[string]*matrix.Matrix
+}
+
+func New(p *plugin.AbstractPlugin) plugin.Plugin {
+	return &VolumeAnalytics{AbstractPlugin: p}
+}
+
+var metrics = []string{
+	"dir_bytes_used",
+	"dir_file_count",
+	"dir_subdir_count",
+}
+
+func (v *VolumeAnalytics) Init() error {
+
+	var err error
+
+	if err = v.InitAbc(); err != nil {
+		return err
+	}
+
+	if err = v.initMatrix(); err != nil {
+		return err
+	}
+
+	m := v.Params.GetChildS("MaxDirectoryCount")
+	if m != nil {
+		MaxDirCollectCount = m.GetContentS()
+	}
+
+	timeout, _ := time.ParseDuration(rest.DefaultTimeout)
+	if v.client, err = rest.New(conf.ZapiPoller(v.ParentParams), timeout); err != nil {
+		v.Logger.Error().Stack().Err(err).Msg("connecting")
+		return err
+	}
+
+	if err = v.client.Init(5); err != nil {
+		return err
+	}
+
+	// Assigned the value to currentVal so that plugin would be invoked first time to populate cache.
+	v.currentVal = v.SetPluginInterval()
+	return nil
+}
+
+func (v *VolumeAnalytics) initMatrix() error {
+	v.data = make(map[string]*matrix.Matrix)
+
+	v.data[explorer] = matrix.New(v.Parent+explorer, explorer, explorer)
+
+	for _, v1 := range v.data {
+		v1.SetExportOptions(matrix.DefaultExportOptions())
+	}
+	for _, k := range metrics {
+		err := matrix.CreateMetric(k, v.data[explorer])
+		if err != nil {
+			v.Logger.Warn().Err(err).Str("key", k).Msg("error while creating metric")
+			return err
+		}
+	}
+	return nil
+}
+
+func (v *VolumeAnalytics) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
+
+	cluster, _ := data.GetGlobalLabels().GetHas("cluster")
+	ver := v.client.Cluster().Version
+	clusterVersion := fmt.Sprintf("%d.%d.%d", ver[0], ver[1], ver[2])
```

Comment:
minor but we do this in two places - probably worth adding a method to `Cluster` that does this and refactor both to us